### DISCUSSION TITLE
"Suggestions matching public code" Preference is not used by Copilot chat.

### DISCUSSION THREAD

**Discussion by edunnett** [community/community#74847] [Copilot Conversations]
### Select Topic Area

Question

### Body

I tried Copilot chat asking for a very basic suggestion for Fluent configuration vs Attributes for a standard POCO in VS Code on Mac. It tells me:

>Sorry, the response matched public code so it was blocked. Please rephrase your prompt. [Learn more](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html).

When I go to my preferences (following the steps linked from 'Learn More') to enable the suggestions, I see that they are already enabled. 
<img width="1084" alt="image" src="https://github.com/community/community/assets/83240961/ce3dea83-ae53-40aa-a525-9ad82f039e2a">

I tried disabling and re-enabling and closing my IDE, but whenever I ask a similar question Copilot chat always tells me

>Sorry, the response matched public code so it was blocked. Please rephrase your prompt. [Learn more](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html).

I only have one account with access to Github copilot, and it is not through an organization. At this point, 30 minutes have elapsed, so I have also tried waiting. 

Is the documentation missing steps, have I not followed them correctly, or could this be a bug?

**Comment by AidenJamesYt** [comment]
It sounds like you may be experiencing a bug with GitHub Copilot Chat.

It's possible that the documentation is missing some steps, but it's also possible that there's a problem with the way that Copilot Chat is filtering public code. I've tried to reproduce the issue on my own, but I haven't been able to.

I recommend that you report the issue to github support. They should be able to investigate the issue and provide you with a workaround or a fix.

In the meantime, you may be able to get around the issue by rephrasing your prompt. For example, instead of asking "Fluent configuration vs Attributes for a standard POCO in VS Code on Mac?", you could ask "What are the pros and cons of using Fluent configuration vs Attributes for a standard POCO in a .NET application?".

You could also try using the Copilot Chat feedback button to let GitHub know that you're experiencing the issue. This will help them to prioritize fixing the bug.

Also, here are some more things you can try:

Make sure that you're using the latest version of Copilot Chat.
Try restarting your IDE.
Try clearing your cache.
Try disabling and re-enabling Copilot Chat.
Try disabling and re-enabling the public code filter.

If you're still having problems, please contact the github support team by opening up a ticket. You can use [this link to open one](https://docs.github.com/en/support/contacting-github-support/creating-a-support-ticket#creating-a-support-ticket)

**Comment by cococococococo** [comment]
I have the same issue

**Comment by edunnett** [comment]
Thanks for your help Aiden. I tried those things with no success. (I wasn't sure which cache to clear though). I also opened a ticket, but it was closed due to the feature being in preview.

> Thank you for reaching out about the technical previews for Copilot! Please note that GitHub Support will not be able to answer questions regarding features under technical preview, including Copilot Chat. 

If anyone else is having this issue I think this means that they know about it and the solution is just to wait.

**Reply by tmichals** [reply]
apparently Co-pilot is down!

**Reply by quinnypig** [reply]
Wow. I seriously confess that this hadn't occurred to me! It really does seem like "response matched public code" is a fallthrough default error message that really sends users down the wrong path...

**Comment by neljaouh** [comment]
I have the same issue, uninstalled and reinstalled the extension but the issue persists ... it's just sad cause it renders Copilot useless to me. The dev team need to do something. In the main time I am back to copy past mode.

**Comment by ishensarabia** [comment]
Having this issue too.

**Comment by TeeDub89** [comment]
it's giving me this response no matter what prompt i give it. this is supposed to be a paid service but i get better results from free chatGPT

**Reply by ishensarabia** [reply]
Seems like it's happening now to everyone

**Reply by mikefaus** [reply]
yeah me too

**Comment by yeelauren** [comment]
Same here, maybe the api is down. From my logs:
```
2023-11-21T20:12:15.776Z [INFO] [chat fetch] Request ID for failed request: , abcdefg

2023-11-21T20:12:15.778Z [ERROR] [chat fetch] Unhandled status from server:,400,Bad Request
```

**Reply by mrtzpngrtz** [reply]
I subscribed to copilot 15 minutes ago, I thought I was getting crazy because google search didn't find anything yet about "Sorry, the response matched public code so it was blocked. Please rephrase your prompt."

**Reply by jcopperman** [reply]
Same here. Found some solace that I'm not the only one

**Comment by jcopperman** [comment]
Copilot is operational on Github status https://www.githubstatus.com/

**Comment by lbold27** [comment]
got the same issue

**Comment by cavricks** [comment]
Same

**Comment by chrismacke** [comment]
Same issue. Just started today.

**Comment by WattersIV** [comment]
Bump

**Comment by andrewmcconville** [comment]
+1. Also experiencing this no matter what I input as a prompt.

**Comment by tmichals** [comment]
me too.. something is broken!

**Comment by np-allane** [comment]
+1

**Comment by Maystra** [comment]
Same for me

**Comment by mikefaus** [comment]
Ok I thought I was going crazy. Apparently, I am not the only one.

**Comment by luis-slyfield** [comment]
+1

**Comment by HectorSaldesFI** [comment]
I got the same ...

**Reply by AndyAndDevid** [reply]
It works again :)

**Comment by AshlinR** [comment]
I'm going for a walk :(

**Comment by CorwinCodes** [comment]
+1

**Comment by jouhrocha** [comment]
Mismo problema miles de personas sin servicio esto merece una explicación y el reintegro de parte de la mensualidad, que seguro que ahí funcionan las pasarelas de pago al 100% eso no se os cae mamo...nes

**Comment by rmzluish** [comment]
hey hey calm down, Sam Altman's Team is working, be cool.

**Comment by HectorSaldesFI** [comment]
Now.. Copilot is working fine on my VSCode :heavy_check_mark:

**Comment by CorwinCodes** [comment]
Back up for me. Co-pilot would benefit from a more straightforward "We're down" message. No reason to throw it back on the user and waste their time.

**Comment by tmichals** [comment]
yes, it would be nicer, but this is pretty sweet already.  bound to be somethings they missed thank goodness they are not SkyNet yet, but that is coming sooner than later I think.
    On Tuesday, November 21, 2023 at 04:44:48 PM EST, CorwinCodes ***@***.***> wrote:  
 
 


Back up for me. Co-pilot would benefit from a more straightforward "We're down" message. No reason to throw it back on the user and waste their time.

**Comment by Sebasxs** [comment]
Same issue here

**Comment by OmerYilmazlar** [comment]
Yeah I have the same problem here.

I allowed it, and waited for 1 whole day.

Still. the issue persists
![image](https://github.com/community/community/assets/58121818/bdebd48c-aa19-443c-9cd4-a69168e289f3)

**Comment by all-mind** [comment]
same issue, setting set to allow but change occurs after restarting, uninstalling, disabling.

**Comment by mkals** [comment]
I have the same issue

**Comment by jrh8182** [comment]
I also have the same problem, how to fix it？

**Comment by gyuray-dev** [comment]
I am still encountering this error in the VS Code Copilot chat. Neither allowing policy, re-installing, signing out/in, nor clearing seems to work.

**Comment by zuggernautt** [comment]
same issue

**Comment by Lafawnduh** [comment]
Still happening for me on VScode Github Copilot Chat extension v0.11.1 + Github Copilot extension v1.153.0

**Comment by gyuray-dev** [comment]
I have discovered that rejections by content filtering also display a 'public code matching' message in the current release version. In my case, the codebase contained the word 'slave' due to its relation to master-slave network communication. You can check the exact error message with VScode Insider and Copilot pre-release version.

This link leads to a related issue: [Public code being blocked when setting is Public code: Allow.](https://github.com/microsoft/vscode-copilot-release/issues/742)

**Comment by quinnypig** [comment]
"Publicly available code." Well it's the React "hello world" demo app I'm trying to work with so I freaking hope it matches what other people have done. If I'm breaking new ground with this thing then we're all in serious trouble.

**Comment by christianbueno1** [comment]
It is the second time about a prompt using GitHub Copilot in vscode.

![image](https://github.com/community/community/assets/10623247/e9909bd6-eb0f-46d1-88c5-bfd48fbeb6c8)

And I have selected "Allowed"

![image](https://github.com/community/community/assets/10623247/152d9df8-493f-49f0-96fe-830d45720e8e)

the prompt: 
In javascript how can we add in a string, literal string , with break line, in this case to assign to the html key in an object of a trasnporter.sendMail function.

**Comment by Robert-Lu** [comment]
Same +1

I want to make a summary of the file to put it on top with chat. Always run into this. I have set allow, also tried re-enable.

**Comment by DanielGraysonPark** [comment]
I'm getting the same issue, only had copilot for a week , is this issue supposed to be fixed now?

**Comment by DanielGraysonPark** [comment]
im getting the same issue, only had copilot for a week , is this issue supposed to be fixed now?

**Comment by andy-dendy-repo** [comment]
same issue

**Comment by Sebasxs** [comment]
In my case, the issue was that I was attempting to use Copilot from an MD file, which is not supported

**Comment by DanShmirer** [comment]
Getting the same issue while using Copilot for business, it is disabled but Devs still getting this error.

**Comment by nm-free** [comment]
I encountered the same issue while using Copilot for business purposes.

Allowed by the administrator, but blocked on individual developer accounts

**Comment by ethus3h** [comment]
I got this message trying to use Copilot to revise public example code to troubleshoot a bug. I think there are a few adjustments that could help the UX around this:

1. If the matching code is present in my prompt, that particular match should be disregarded, as it's just repeating what I'd entered myself.
2. After getting the message, there isn't a "bypass once" option. I usually want that setting turned on, but in cases like this it's not helpful. Instead, I lost the generated response, and I guess because I navigated away from the page, I lost my prompt also.
    2b. I saw it generate the top part of the response, before it was replaced by the 'public code' message. That first part would have been enough for me to work with and finish the rest myself. It would be nice to truncate the response to the non-matching part, so I could still use that much.
4. After changing the setting to Allowed, it showed the 'public code' error message again, as described in this issue.
